examples/long_running.py

```diff
@@ -0,0 +1,121 @@
+"""Example specification for a consumption-savings model with health and leisure."""
+
+import jax.numpy as jnp
+
+# ======================================================================================
+# Numerical parameters and constants
+# ======================================================================================
+N_GRID_POINTS = {
+    "states": 100,
+    "choices": 200,
+}
+
+RETIREMENT_AGE = 65
+
+# ======================================================================================
+# Model functions
+# ======================================================================================
+
+
+# --------------------------------------------------------------------------------------
+# Utility function
+# --------------------------------------------------------------------------------------
+def utility(consumption, working, health, sport, delta):
+    return jnp.log(consumption) - (delta - health) * working - sport
+
+
+# --------------------------------------------------------------------------------------
+# Auxiliary variables
+# --------------------------------------------------------------------------------------
+def working(leisure):
+    return 1 - leisure
+
+
+def wage(age):
+    return 1 + 0.1 * age
+
+
+def age(_period):
+    return _period + 18
+
+
+# --------------------------------------------------------------------------------------
+# State transitions
+# --------------------------------------------------------------------------------------
+def next_wealth(wealth, consumption, working, wage, interest_rate):
+    return (1 + interest_rate) * (wealth - consumption) + wage * working
+
+
+def next_health(health, sport, working):
+    return health * (1 + sport - working / 2)
+
+
+def next_wealth_with_shock(
+    wealth,
+    consumption,
+    working,
+    wage,
+    wage_shock,
+    interest_rate,
+):
+    return interest_rate * (wealth - consumption) + wage * wage_shock * working
+
+
+# --------------------------------------------------------------------------------------
+# Constraints
+# --------------------------------------------------------------------------------------
+def consumption_constraint(consumption, wealth):
```

Comment:
```suggestion
def consumption_constraint(consumption, wealth, labor_income):
```
Else states with 0 wealth cannot support positive consumption. Probably useful to define a function labor_income, in that case stochastics are put into that function as well, no need to carry around shocks.

Reply:
See below. This actually follows the DC-EGM paper.

Reply:
The we should call it dc_egm example and use a sensible Phelps-Deaton for our own examples!

This basically means your starve to death despite working for a good wage if you have not saved anything...